### PR TITLE
Add opentelemetry

### LINKED ISSUE #201: Add OpenTelemetry integration
See https://github.com/hyperledger/fabric-rfcs/blob/main/text/0000-opentelemetry-tracing.md

Adding OpenTelemetry integration, starting with adding traces support for chaincodes.

### PR DESCRIPTION
**Pull request by atoulme**
Fixes #201

See #153 for previous PR.

Signed-off-by: Antoine Toulme antoine@lunar-ocean.com

This adds OpenTelemetry to Fabric Java chaincode.

### CODE REVIEW

**Comment by atoulme** [comment]
It looks like we're in a good spot. I see one test failure but it seems unrelated. I cannot reproduce the failure locally.

**Comment by atoulme** [comment]
Now the failure is gone - added some more debugging so we can get more info it it ever fails again.

**Comment by denyeart** [comment]
Thanks @atoulme ! Could you summarize the overall plan of attack? In the RFC we discussed it would be better to instrument the peer and client SDKs first, so that you could get an end-to-end view across peer activities (including but not limited to chaincode calls across all the chaincode languages), rather than the view inside a single chaincode language:
https://github.com/hyperledger/fabric-rfcs/pull/42#issuecomment-789491733

**Comment by atoulme** [comment]
The overall plan of attack is to add OpenTelemetry to Fabric - SDKs, because that's where calls come in, peers and orderers, and yes, chaincode so we can track traces. All those components report traces independently and correlation happens after collection in the telemetry tool.

**Comment by denyeart** [comment]
Ok thanks, as we discussed in the RFC, could we start with the peer to figure out the right contracts etc, since the peer is the anchor that ties everything together for a transaction, and then work to the various edges including SDKs and chaincodes. We will also be shifting more towards external chaincode as a service and I don't know if that may impact the chaincode design.

**Review by mbwhite — APPROVED** [review]
LGTM  Thanks

**Comment by atoulme** [comment]
@mbwhite completely understand your curiosity! OpenTelemetry has defined specific environment variables for configuration that can now be passed in as chaincode configuration, environment variables or system properties into the chaincode environment. These are standardized and defined [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md).

The next step is to tie in those configuration elements to the chaincode deployer, so the configuration is passed in.
